### PR TITLE
feat(ci): code checks with mypy and flake8

### PR DESCRIPTION
**Pull request by grdddj**
Including code checks into CI and making an easy `make check` command to check everything locally.

NOTE: in the #58 PR, there is a `tests` folder with mypy testing file, which should be then deleted, as this is much better approach.

### CODE REVIEW

**Comment by grdddj** [comment]
The current CI is failing because of mypy issues, which are already fixed in #58.

Before merging this, we can make a fixup commit here to make it succeed to verify it.

**Comment by grdddj** [comment]
Fixup commit da4eddf:
- Adding a newline at the end of `setup.cfg` file

**Review by tsusanka — APPROVED** [review]
Great :+1:. Since we are merging into `incoming` I do not mind merging a failing PR, rebase #58 on top of it and make sure it passes there. To simplify things. Sounds okay?

**Comment by grdddj** [comment]
> Great +1. Since we are merging into `incoming` I do not mind merging a failing PR, rebase #58 on top of it and make sure it passes there. To simplify things. Sounds okay?

I would agree. Let me squash these two commits into one and force-push here

**Review by vdovhanych — APPROVED** [review]
Looks good

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length = 120
```
Comment (tsusanka) [inline setup.cfg:2]: Btw I am not sure this is needed as `black` formats lines to max 88 (or smth like that).

Comment (grdddj) [reply setup.cfg:2]: Good point! However, when not specified, flake has the upper limit of 79, so it would still scream

Comment (tsusanka) [reply setup.cfg:2]: Ah, funny :). But let's put it to 88 so we are inline with black? It is indeed 88 https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html

Comment (grdddj) [reply setup.cfg:2]: Right, thanks for the source, I will change it.

They even say `If you’re using Flake8, you can bump max-line-length to 88 and mostly forget about it.`

Comment (tsusanka) [reply setup.cfg]: Great :+1:

**Comment by tsusanka** [comment]
I have realized one more nit-pick, sorry. It is slightly better to actually have two jobs in the CI, one for style, one for mypy. Only for the simple reason that you see right away what went wrong. So if you do not mind I would split the `full code check` into `style` and `mypy` jobs.

If you find the full check handy for local development feel free to keep it in the Makefile.

**Comment by grdddj** [comment]
> I have realized one more nit-pick, sorry. It is slightly better to actually have two jobs in the CI, one for style, one for mypy. Only for the simple reason that you see right away what went wrong. So if you do not mind I would split the `full code check` into `style` and `mypy` jobs.
> 
> If you find the full check handy for local development feel free to keep it in the Makefile.

No issue, I can split it. The code_check should probably go first, as it is more important than the style_check.

And yeah, I really like the ability of running simple `make check` locally, so will keep it there

**Comment by grdddj** [comment]
In 9e5e0f6 I tried an experiment with refactoring common information out of the jobs - inspired by [firmware](https://github.com/grdddj/trezor-firmware/blob/master/ci/prebuild.yml#L1)

Turned out the stage needs to belong to every script.

Fixed in e97e2eb

**Review by vdovhanych — APPROVED** [review]
Also, fine with me, I have some nitpicks that there are no newlines at the end of the yml file, but that is no big deal, just my OCD (and yeah, I think I made a mistake when I first committed it here) will add `.editorconfig` file, so we don't have to think about this.

**Review by tsusanka — APPROVED** [review]
Great, feel free to squash and let's merge.

**Comment by tsusanka** [comment]
> No issue, I can split it. The code_check should probably go first, as it is more important than the style_check.

Btw all jobs inside one stage run at the same time (well up to the number of allowed parallel jobs which is 8 ATM I believe).

**Pull request merged**